lib/azure/armrest/armrest_service.rb

```diff
@@ -253,11 +254,20 @@ def rest_head(options)
 
         def raise_api_exception(err)
           begin
-            response = JSON.parse(err.http_body)
-            code     = response['error']['code']
-            message  = response['error']['message']
+            response = case err.response.headers[:content_type]
+            when "application/json"
```

Comment:
Some response headers have Content-Type `"application/json; charset=utf-8"` so we'll need to handle that here